include/lsst/afw/detection/Footprint.h

```diff
@@ -0,0 +1,383 @@
+     * This function returns the real number of pixels, not the area of the bbox.
+     */
+    size_t getArea() const { return _spans->getArea(); }
+
+    /**
+     * @brief Return the Footprint's centroid
+     *
+     * The centroid is calculated as the mean of the pixel centers
+     */
+    geom::Point2D getCentroid() const { return _spans->computeCentroid(); }
+
+    /**
+     * @brief Return the Footprint's shape (interpreted as an ellipse)
+     *
+     * The shape is determined by measuring the moments of the pixel
+     * centers about its centroid (cf. getCentroid)
+     */
+    geom::ellipses::Quadrupole getShape() const { return _spans->computeShape(); }
+
+    /**
+     * @brief Shift a Footprint by <tt>(dx, dy)</tt>
+     *
+     * @param dx - How much to move Footprint in column direction
+     * @param dy - How much to move in row direction
+     */
+    void shift(int dx, int dy);
+
+    /**
+    * @brief Shift a Footprint by a given extent
+    *
+    * @param d - ExtentI object which gives the dimensions the Footprint should be shifted
+    */
+    void shift(geom::ExtentI const & d) {shift(d.getX(), d.getY());}
+
+    /**
+     * @brief Return the Footprint's bounding box
+     */
+    geom::Box2I getBBox() const { return _spans->getBBox(); }
+
+    /**
+     * @brief Return the corners of the MaskedImage the footprints live in
+     */
+    geom::Box2I const & getRegion() const { return _region; }
+
+    /**
+     * @brief Set the corners of the MaskedImage wherein the footprints dwell
+     *
+     * @param region - A box describing the corners of the Image the Footprint derives from
+     */
+    void setRegion(geom::Box2I const & region) { _region = region; }
+
+
+    /**
+     * @brief Clip the Footprint such that all values lie inside the supplied Bounding Box
+     *
+     * @param bbox - Integer box object that defines the boundarys the footprint should be
+     *               clipped to.
+     */
+    void clipTo(geom::Box2I const & bbox);
+
+    /**
+     * @brief Tests if a pixel postion falls inside the Footprint
+     *
+     * @param pix - Integer point object defining the position of a pixel to test
+     */
+    bool contains(geom::Point2I const & pix) const;
+
+    /**
+     *  @brief Transform the footprint from one WCS to another
+     *
+     *  @param source - Wcs that defines the coordinate system of the input footprint.
+     *  @param target - Wcs that defines that desired coordinate system of the returned footprint.
+     *  @param region - Used to set the "region" box of the returned footprint; note that this is
+     *                  NOT the same as the footprint's bounding box.
+     *  @param doClip - If true, clip the new footprint to the region bbox before returning it.
+     */
+    std::shared_ptr<Footprint> transform(
+        std::shared_ptr<image::Wcs> source,
+        std::shared_ptr<image::Wcs> target,
+        geom::Box2I const & region,
+        bool doClip=true
+    ) const;
```

Comment:
While we may want this for backwards compatibility, we probably want the `transform` interfaces that `SpanSet` has too.